### PR TITLE
Implement new extension preferences

### PR DESCRIPTION
**Pull request by codeimpl**
The Banana Extension list is getting longer and needs reorganization.
This patch reorganized the list by dividing it into detailed settings
and quick settings.

Refs: #809
Co-authored-by: Jaeyong Bae <jdragon.bae@gmail.com>

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -334,7 +339,12 @@ android_resources("triple_banana_resources") {
     "//triple_banana/res/values/dimens.xml",
     "//triple_banana/res/values/strings.xml",
     "//triple_banana/res/values/styles.xml",
+    "//triple_banana/res/xml/advanced_features_preferences.xml",
     "//triple_banana/res/xml/banana_extension_preferences.xml",
+    "//triple_banana/res/xml/media_features_preferences.xml",
+    "//triple_banana/res/xml/new_banana_extension_preferences.xml",
+    "//triple_banana/res/xml/privacy_features_preferences.xml",
+    "//triple_banana/res/xml/appearance_features_preferences.xml",
```
Comment (jdragonbae) [inline BUILD.gn:347]: We need to `git cl format`.

File: res/xml/new_banana_extension_preferences.xml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/adblock_summary"
+        android:title="@string/adblock_with_version" />
+
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_secure_login"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/secure_login_summary"
+        android:title="@string/secure_login" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:23]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_secure_dns"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/secure_dns_summary"
+        android:title="@string/secure_dns" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:29]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_background_play"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/background_play_summary"
+        android:title="@string/background_play" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:35]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_dark_mode"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/dark_mode_summary"
+        android:title="@string/dark_mode" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:41]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_translate"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/languages_offer_translate_switch"
+        android:title="@string/translate_button" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:47]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_browser_lock"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/browser_lock_summary"
+        android:title="@string/browser_lock" />
+
+    <PreferenceCategory android:title="@string/detail_settings" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:55]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    <Preference
+        android:fragment="org.triple.banana.settings.AdvancedFeatures"
+        android:summary="@string/advanced_features_summary"
+        android:title="@string/advanced_features" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:60]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    <Preference
+        android:fragment="org.triple.banana.settings.MediaFeatures"
+        android:summary="@string/media_features_summary"
+        android:title="@string/media_features" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:65]: No blank

File: res/xml/new_banana_extension_preferences.xml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    <Preference
+        android:fragment="org.triple.banana.settings.AppearanceFeatures"
+        android:summary="@string/appearance_summary"
+        android:title="@string/appearance" />
+
```
Comment (romandev) [inline res/xml/new_banana_extension_preferences.xml:70]: No blank

File: res/xml/privacy_features_preferences.xml [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+<!-- Copyright 2020 The Triple Banana Authors. All rights reserved.
+     This Source Code Form is subject to the terms of the Mozilla Public
+     License, v. 2.0. If a copy of the MPL was not distributed with this
+     file, You can obtain one at https://mozilla.org/MPL/2.0/. -->
+
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orderingFromXml="false">
+
+    <PreferenceCategory android:title="@string/privacy" />
+
```
Comment (romandev) [inline res/xml/privacy_features_preferences.xml:11]: No blank

File: res/xml/privacy_features_preferences.xml [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_secure_login"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/secure_login_summary"
+        android:title="@string/secure_login" />
+
```
Comment (romandev) [inline res/xml/privacy_features_preferences.xml:17]: No blank

File: res/xml/privacy_features_preferences.xml [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_secure_dns"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/secure_dns_summary"
+        android:title="@string/secure_dns" />
+
```
Comment (romandev) [inline res/xml/privacy_features_preferences.xml:23]: No blank

File: res/xml/privacy_features_preferences.xml [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="feature_name_browser_lock"
+        android:summaryOff="@string/text_off"
+        android:summaryOn="@string/browser_lock_summary"
+        android:title="@string/browser_lock" />
+
+    <PreferenceCategory android:title="@string/user_data" />
+
```
Comment (romandev) [inline res/xml/privacy_features_preferences.xml:31]: No blank

File: res/xml/privacy_features_preferences.xml [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+    <Preference
+        android:fragment="org.chromium.chrome.browser.browsing_data.ClearBrowsingDataTabsFragment"
+        android:key="clear_browsing_data"
+        android:summary="@string/clear_browsing_data_summary"
+        android:title="@string/clear_browsing_data_title" />
+
```
Comment (romandev) [inline res/xml/privacy_features_preferences.xml:37]: No blank

File: src/org/triple/banana/settings/AppearanceFeatures.java [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+package org.triple.banana.settings;
+
+import android.os.Bundle;
+
+import androidx.preference.PreferenceFragmentCompat;
+
+import org.triple.banana.R;
+
+public class AppearanceFeatures extends PreferenceFragmentCompat {
```
Comment (romandev) [inline src/org/triple/banana/settings/AppearanceFeatures.java:14]: AppearanceSettings

File: src/org/triple/banana/settings/MediaFeatures.java [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+
+package org.triple.banana.settings;
+
+import android.os.Bundle;
+
+import androidx.preference.PreferenceFragmentCompat;
+
+import org.triple.banana.R;
+
+public class MediaFeatures extends PreferenceFragmentCompat {
```
Comment (romandev) [inline src/org/triple/banana/settings/MediaFeatures.java:14]: MediaFeatureSettings

File: src/org/triple/banana/settings/PrivacyFeatures.java [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+
+package org.triple.banana.settings;
+
+import android.os.Bundle;
+
+import androidx.preference.PreferenceFragmentCompat;
+
+import org.triple.banana.R;
+
+public class PrivacyFeatures extends PreferenceFragmentCompat {
```
Comment (romandev) [inline src/org/triple/banana/settings/PrivacyFeatures.java:14]: PrivacySettings

File: BUILD.gn [hunk 19]
```diff
@@ -334,7 +339,12 @@ android_resources("triple_banana_resources") {
     "//triple_banana/res/values/dimens.xml",
     "//triple_banana/res/values/strings.xml",
     "//triple_banana/res/values/styles.xml",
+    "//triple_banana/res/xml/advanced_features_preferences.xml",
+    "//triple_banana/res/xml/appearance_features_preferences.xml",
```
Comment (romandev) [inline BUILD.gn:343]: appearance_preferences

File: BUILD.gn [hunk 20]
```diff
@@ -334,7 +339,12 @@ android_resources("triple_banana_resources") {
...
     "//triple_banana/res/xml/banana_extension_preferences.xml",
+    "//triple_banana/res/xml/media_features_preferences.xml",
+    "//triple_banana/res/xml/new_banana_extension_preferences.xml",
+    "//triple_banana/res/xml/privacy_features_preferences.xml",
```
Comment (romandev) [inline BUILD.gn:347]: privacy_preferences

File: BUILD.gn [hunk 21]
```diff
@@ -203,8 +203,13 @@ android_library("triple_banana_java") {
     "//triple_banana/src/org/triple/banana/secure_dns/SecureDnsNotificationManager.java",
 
     # settings
+    "//triple_banana/src/org/triple/banana/settings/AdvancedFeatures.java",
```
Comment (romandev) [inline BUILD.gn:206]: AdvancedFeatureSettings

File: BUILD.gn [hunk 22]
```diff
@@ -203,8 +203,13 @@ android_library("triple_banana_java") {
...
+    "//triple_banana/src/org/triple/banana/settings/AppearanceFeatures.java",
```
Comment (romandev) [inline BUILD.gn:207]: AppearanceSettings

File: BUILD.gn [hunk 23]
```diff
@@ -203,8 +203,13 @@ android_library("triple_banana_java") {
...
     "//triple_banana/src/org/triple/banana/settings/ExtensionFeatures.java",
     "//triple_banana/src/org/triple/banana/settings/LongClickableSwitchPreference.java",
+    "//triple_banana/src/org/triple/banana/settings/MediaFeatures.java",
```
Comment (romandev) [inline BUILD.gn:210]: MediaFeatureSettings

File: BUILD.gn [hunk 24]
```diff
@@ -203,8 +203,13 @@ android_library("triple_banana_java") {
...
+    "//triple_banana/src/org/triple/banana/settings/NewExtensionFeatures.java",
+    "//triple_banana/src/org/triple/banana/settings/PrivacyFeatures.java",
```
Comment (romandev) [inline BUILD.gn:212]: PrivacyFeatureSettings

File: BUILD.gn (hunk 21 above)
Comment (jdragonbae) [reply BUILD.gn:206]: Done.

**Review by romandev — APPROVED** [review]
lgtm

**Pull request merged**